### PR TITLE
Don't compute "hash data" for options that are not used by Bloop

### PR DESCRIPTION
**Pull request by alexarchambault**
This PR makes that more explicit via `notForBloopOptions: PostBuildOptions` in `BuildOptions`.

This hash is only used to make Bloop directories unique, so it doesn't need to take package or repl options into account.

### CODE REVIEW

File: modules/build/src/main/scala/scala/build/options/BuildOptions.scala [hunk 1]
```diff
@@ -38,8 +38,7 @@ final case class BuildOptions(
   internal: InternalOptions = InternalOptions(),
   mainClass: Option[String] = None,
   testOptions: TestOptions = TestOptions(),
-  packageOptions: PackageOptions = PackageOptions(),
-  replOptions: ReplOptions = ReplOptions()
+  notForBloopOptions: PostBuildOptions = PostBuildOptions()
```
Comment (romanowski) [inline modules/build/src/main/scala/scala/build/options/BuildOptions.scala:41]: I think we should call it `postBuildOptions`

**Review by romanowski — APPROVED** [review]

File: modules/build/src/main/scala/scala/build/options/BuildOptions.scala (hunk 1 above)
Comment (alexarchambault) [reply modules/build/src/main/scala/scala/build/options/BuildOptions.scala:41]: Or maybe rename `PostBuildOptions` to `NotForBloopOptions `? I wanted to make it clear that it shouldn't be used when generating the Bloop config for example.

Comment (romanowski) [reply modules/build/src/main/scala/scala/build/options/BuildOptions.scala:41]: It is strange that BuildOptions contains PostBuild / NotForBloop. maybe we should Refactor it at some point so we have an Option object that contains BuildOptions (ones that bloop cares for), runtime options (like main class, tests, repl things), package options etc.

Comment (alexarchambault) [reply modules/build/src/main/scala/scala/build/options/BuildOptions.scala:41]: That's what I wanted to do initially, something like
```scala
final case class AllBuildOptions(
  buildOptions: BuildOptions = BuildOptions(),
  postBuildOption: PostBuildOptions = PostBuildOptions()
)
```

I just fear it'll touch many-many files in the codebase.

Comment (romanowski) [reply modules/build/src/main/scala/scala/build/options/BuildOptions.scala:41]: ok, we can live now with `notForBloopOptions` and refactor option to `AlLBuildOption` at some point

**Pull request merged**